ste/sender-azureFile.go

```diff
@@ -450,21 +451,25 @@ func (u *azureFileSenderBase) SetFolderProperties() error {
 		return err
 	}
 
-	_, err = u.getDirectoryClient().SetMetadata(u.ctx, &directory.SetMetadataOptions{Metadata: u.metadataToApply})
-	if err != nil {
-		return err
-	}
-
 	err = u.DoWithOverrideReadOnly(u.ctx,
 		func() (interface{}, error) {
+			_, err := u.getDirectoryClient().SetMetadata(u.ctx, &directory.SetMetadataOptions{Metadata: u.metadataToApply})
+			if err != nil {
+				return nil, err
+			}
 			return u.getDirectoryClient().SetProperties(u.ctx, &directory.SetPropertiesOptions{
 				FileSMBProperties: &u.smbPropertiesToApply,
 				FilePermissions:   &u.permissionsToApply,
 			})
 		},
 		u.fileOrDirClient,
 		u.jptm.GetForceIfReadOnly())
-	return err
+
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
You can directly return err here.
```suggestion
	return err
```

Reply:
Done in the latest commit, thanks for the suggestion